### ISSUE TITLE
Variable 'derive' flag should be settable at project level

### ISSUE THREAD

**Issue by senesis** [ESMValGroup/ESMValCore#1086]
According to [the relevant doc section](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/recipe/preprocessor.html#variable-derivation), using a derived variable needs to declare it as derived in the 'variable' section of the recipe, using flag 'derive', and according to _recipe._add_cmor_info, default value for the derive flag is False.

This does not fit a need which is slightly beyond the scheme author's goal (which was : defining a variable which is not in the CMIP standard data request), but which is a use case need. This is the case of a model which outputs are described in some project which itself has e.g. cmor_type=CMIP6, but which does not provide some CMIP6 variable. 

With the present scheme this implies to modify the 'diagnostics' section of any recipe involving that variable and that one wants to apply to that model data (for setting the derive flag to True), while this 'diagnostics' section should be as independent as possible of the 'datasets' section , and specifically independent of the very list of datasets used.

My suggestion would be to change that line of _recipe._add_cmor_info :

>     derive = variable.get('derive', False)

to allow the default value to be set by querying some entry of the project config

One could even argue that the default value in the project's config should have higher priority than the one in the recipe's 'variable' section

**Comment by bouweandela** [comment]
Note that you can add `derive: true` to the dataset entry in the dataset definition instead of in the variable definition. This is explained [here](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/recipe/overview.html#variable-and-dataset-definitions) in the documentation. For example:

```yaml
datasets:
  - {dataset: BCC-ESM1, project: CMIP6, exp: historical, ensemble: r1i1p1f1, grid: gn}
  - {dataset: CanESM2, project: CMIP5, exp: historical, ensemble: r1i1p1, derive: true}

diagnostics:

  map:
    description: Global map of temperature in January 2000.
    themes:
      - phys
    realms:
      - atmos
    variables:
      alb:
        mip: Amon
        preprocessor: select_january
        start_year: 2000
        end_year: 2000
    scripts:
      script1:
        script: examples/diagnostic.py
        quickplot:
          plot_type: pcolormesh
          cmap: Reds
```

would enable variable derivation only for the CanESM2 dataset, but not for BCC-ESM1 and there is no need to set this in the variable section of the recipe.

**Comment by senesis** [comment]
> Note that you can add `derive: true` to the dataset entry in the dataset definition instead of in the variable definition. This is explained [here](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/recipe/overview.html#variable-and-dataset-definitions) in the documentation. 

Fine, but assume that you want to use a recipe that uses two variables, and that, for your model/project, only one should go through the derive process. You would then implicitly put the derive flag 'on' also for that other one, which don't need it. 

That's OK only if there is strictly no side effect for this other variable. And if the same applies to the 'force_derived' flag, there should be at least the side effect of additional computation for the variable which is OK 

But I agree that this somewhat answers my concern, namely to avoid changing the diagnostics part of the recipe

**Comment by bouweandela** [comment]
If the variable is available, it will not be derived. Only if you also set the `force_derived` flag.

**Comment by bouweandela** [comment]
With the additional configuration option implemented in #1124 there will be the possibility to set this flag automatically per dataset and per variable without modifying the recipe.

**Comment by senesis** [comment]
Yes, that will be fine !
